File: deamhome/src/main/java/sosteam/deamhome/domain/popup/entity/PopUp.java
package sosteam.deamhome.domain.popup.entity;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import sosteam.deamhome.domain.account.entity.Account;
import sosteam.deamhome.global.entity.BaseEntity;

import java.time.LocalDateTime;

@Document
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PopUp extends BaseEntity {

    @NotNull
    private String title;

    @NotNull
    private String content;

    @NotNull
    private String ip;

    //팝업 활성화 여부
    @NotNull
    private boolean isOnBoard;

    @NotNull
    @Setter
    private LocalDateTime endAt;

    @DBRef
    @Setter
    private Account account;
}

File: src/main/kotlin/sosteam/deamhome/domain/popup/entity/PopUp.kt
package sosteam.deamhome.domain.popup.entity

import jakarta.validation.constraints.NotNull
import lombok.Setter
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import sosteam.deamhome.domain.account.entity.Account
import sosteam.deamhome.global.entity.BaseEntity
import sosteam.deamhome.global.image.entity.Image
import java.time.LocalDateTime

@Table("pop_up")
class PopUp(
	@Id
	var id: Long?,
	var title: String,
	var createdIp: String,
	
	//팝업 활성화 여부
	var isOnBoard: @NotNull Boolean = false,
	
	var endAt: LocalDateTime,
	
	image: Image,
	account: Account
) : BaseEntity() {
	
//	@DocumentReference
//	@Setter
//	var image: Image = image
//
//	@DocumentReference
//	@Setter
//	val account: Account = account
}